core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -473,10 +473,26 @@ public boolean delete(Object key) {
          resizeLock.readLock().unlock();
       }
 
+      org.infinispan.commons.io.ByteBuffer keyBb;
+      org.infinispan.commons.io.ByteBuffer valueBb = null;
```

Comment:
If null is passed for both values, then we get an "ambiguous method call" error.  I passed valueBb and metadataBb to avoid casting for both values.